File: miscellaneous/run_length_encoding.cpp
//AlgoExpert - run-length encoding - strings

#include<vector>
using namespace std;

string rle(string str){
	vector<char> esc;
	int crl = 1;
	for(int i = 1; i < str.size(); i++){
		char cc = str[i];
		char pc = str[i - 1];
		if (cc != pc || crl == 9){
			esc.push_back(to_string(crl)[0]);
			esc.push_back(pc);
			crl = 0;
		}
		crl ++;
	}
	esc.push_back(to_string(crl)[0]);
	esc.push_back(str[str.size() - 1]);
	string es(esc.begin(),esc.end());
	return es;
}

File: miscellaneous/run_length_encoding.py
'''
AlgoExpert - run-length encoding - strings
'''

def runLengthEncoding(string):
	esc = []
	crl = 1
	for i in range(1, len(string)):
		cc = string[i]
		pc = string[i - 1]
		
		if cc != pc or crl == 9:
			esc.append(str(crl))
			esc.append(pc)
			crl = 0
		
		crl += 1
	
	esc.append(str(crl))
	esc.append(string[len(string) - 1])
	
	return "".join(esc)
